In [ ]:
# import necessary libraries

import numpy as np # for linear algebra
import pandas as pd # for creating dataframes

from statsmodels.formula.api import ols # for Ordinary Least Squares method

In [3]:
# import the dataset

survey_data = pd.read_csv('dataSurvey.csv')

In [ ]:
# preview the dataset

survey_data.head(5)

,Rating,attendance,avg_test_score,sleep_hours,Unnamed: 4
0,65,50,70,6,NaN
1,93,95,90,8,NaN
2,5,6,7,8,NaN
3,92,80,90,6,
4,90,85,90,6,NaN
5,95,95,90,9,NaN
6,90,70,89,5,NaN
7,91,90,92,8,NaN
8,98,95,95,3,NaN
9,94,80,94,2,NaN


In [ ]:
#  drop unnecessary columns (if needed)

survey_data = survey_data.drop(columns='Unnamed: 4')

In [ ]:
# preview the dataset again

survey_data.head(5)

,Rating,attendance,avg_test_score,sleep_hours
0,65,50,70,6
1,93,95,90,8
2,5,6,7,8
3,92,80,90,6
4,90,85,90,6
5,95,95,90,9
6,90,70,89,5
7,91,90,92,8
8,98,95,95,3
9,94,80,94,2


In [10]:
# specify the regression with OLS

model_for_survey = ols(formula='Rating~attendance+avg_test_score+sleep_hours', data=survey_data).fit()

In [ ]:
# display the parameters of the model 
# add '.round(3)' after 'params' to have them rounded to 3 decimal places

model_for_survey.params

Intercept        -24.907255
attendance        -0.592227
avg_test_score     1.637518
sleep_hours        1.964732
dtype: float64

# **OUR REGRESSION**

## **Rating** = -24.907255 + (-0.592227)\***attendance** + 1.637518\***avg_test_score** + 1.964732\***sleep_hours**